src/readcalls.c

```diff
@@ -81,322 +80,128 @@ void init_scoring(void) {
-
-	    strcpy(hiscall, qso->call);
-
-	    add_ok = pacc_pa();
-
-	    if (add_ok == false) {
-		qsos_per_band[bandindex]++;
-	    }
-
-	    hiscall[0] = '\0';
+	dupe = is_dupe(qso->call, qso->bandindex, qso->mode);
+	addcall(qso);
+	score_qso();    //FIXME argument?
+
+	char *logline = makelogline(qso);
+
+	if (strcmp(logline, qsos[linenr - 1]) != 0) {
+	    // different: update log line and mark change
+	    g_strlcpy(qsos[linenr - 1], logline, sizeof(qsos[0]));
+	    g_free(qso->logline);
+	    qso->logline = g_strdup(logline);
+	    log_changed = true;
 	}
 
-	if (pfxnummultinr > 0) {
-	    getpx(qso->call);
-	    pxnr = districtnumber(wpx_prefix);
+	g_free(logline);
 
-	    getctydata(qso->call);
-
-	    pfxnumcntidx = lookup_country_in_pfxnummult_array(countrynr);
-
-	    add_ok = true;
-	}
-
-	veto = check_veto(countrynr);
-
-	if (add_ok) {
-
-	    worked[station].band |= inxes[bandindex];	/* mark band as worked */
-
-	    qsos_per_band[bandindex]++;
-
-	    if (CONTEST_IS(CQWW) || itumult || wazmult)
-		zones[z] |= inxes[bandindex];
-
-	    if (pfxnumcntidx < 0) {
-		if (!veto) {
-		    countries[countrynr] |= inxes[bandindex];
-		}
-	    } else {
-		pfxnummulti[pfxnumcntidx].qsos[pxnr] |= inxes[bandindex];
-	    }
-	}
-	free_qso(qso);
-	qso = NULL;
+	g_ptr_array_add(qso_array, qso);
     }
 
     fclose(fp);
 
-    /* all lines red, now build other statistics */
-    if (CONTEST_IS(WPX) || pfxmult) {
-	char checkcall[20];
-
-	/* build prefixes_worked array from list of worked stations */
-	InitPfx();
-
-	for (int n = 0; n < nr_worked; n++) {
-	    strcpy(checkcall, worked[n].call);
-	    getpx(checkcall);
-	    /* FIXME: wpx is counting pfx only once so bandindex is not
-	     * really needed here. If you have wpx and pfxmultab set the
-	     * count for the last read band wil be wrong as all pfx will be
-	     * counted for that band.
-	     * Maybe better use BANDINDEX_OOB here:
-	     * - Will count pfx for wpx correctly
-	     * - but will not change counts for pfxmultab on contest bands */
-	    add_pfx(wpx_prefix, BANDINDEX_OOB);
-	}
-    }
-
-    if (CONTEST_IS(CQWW) || itumult || wazmult) {
-	for (int n = 1; n < MAX_ZONES; n++) {
-	    count_contest_bands(zones[n], zonescore);
-	}
-    }
-
-    if (CONTEST_IS(CQWW)) {
-	for (int n = 1; n <= MAX_DATALINES - 1; n++) {
-	    count_contest_bands(countries[n], countryscore);
-	}
-    }
-
-    if (dx_arrlsections) {
-	for (int cntr = 1; cntr < MAX_DATALINES; cntr++) {
-	    if (cntr != w_cty && cntr != ve_cty) {	// W and VE don't count here...
-		count_contest_bands(countries[cntr], countryscore);
+    if (log_changed && interactive) {
+	showmsg("Log changed due to rescoring. Do you want to save it? Y/(N)");
+	if (toupper(key_get()) == 'Y') {
+	    // save a backup
+	    char prefix[40];
+	    format_time(prefix, sizeof(prefix), "%Y%m%d_%H%M%S");
+	    char *backup = g_strdup_printf("%s_%s", prefix, logfile);
+	    rename(logfile, backup);
+	    // rewrite log
+	    remove(logfile);
```

Comment:
There should be no 'logfile' after the rename.

Reply:
fixed